src/back/wgsl/writer.rs

```diff
@@ -880,13 +880,22 @@ impl<W: Write> Writer<W> {
                 write!(self.out, "{}", level)?;
                 write!(self.out, "switch(")?;
                 self.write_expr(module, selector, func_ctx)?;
+                let uint = func_ctx.info[selector]
+                    .ty
+                    .inner_with(&module.types)
+                    .scalar_kind()
+                    == Some(crate::ScalarKind::Uint);
                 writeln!(self.out, ") {{")?;
 
                 // Write all cases
                 let mut write_case = true;
                 let all_fall_through = cases
                     .iter()
                     .all(|case| case.fall_through && case.body.is_empty());
+                let postfix = || match uint {
```

Comment:
using closures here seems like an overkill. Couldn't we just do:
```rust
let type_postfix = match func_ctx.info[selector].ty.inner_with(&module.types) {
  crate::TypeInner::Scalar { kind: crate::ScalarKind::Uint, .. } => "u",
  _ => "",
};
```
I think doing this in other places would also improve things a little.

Reply:
ok I realized what you meant now